public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistoryScene.tsx

```diff
@@ -189,71 +181,72 @@ function getSceneQuery(datasource: DataSourceInformation) {
+        .setCustomFieldConfig('stacking', { mode: StackingMode.None, group: 'A' })
+        .setCustomFieldConfig('gradientMode', GraphGradientMode.Hue)
+        .setCustomFieldConfig('scaleDistribution', { type: ScaleDistribution.Linear })
+        .setOption('legend', { showLegend: false, displayMode: LegendDisplayMode.Hidden })
+        .setOption('tooltip', { mode: TooltipDisplayMode.Single })
+        .setNoValue('No events found')
+        .build() as VizPanel<Options, FieldConfig>,
+  });
+}
+
+export class ClearFilterButtonScenesObject extends SceneObjectBase {
+  public static Component = ClearFilterButtonObjectRenderer;
 
-      .setNoValue('No events found')
-      .build(),
+  protected _variableDependency = new VariableDependencyConfig(this, {
+    variableNames: [LABELS_FILTER, STATE_FILTER_FROM, STATE_FILTER_TO],
   });
 }
-/*
- * This component shows a button to clear the filters.
- * It is shown when the filters are active.
- * props:
- * labelsFilterVariable: the textbox variable for filtering by labels
- * transitionsToFilterVariable: the custom variable for filtering by the current state
- * transitionsFromFilterVariable: the custom variable for filtering by the previous state
- */
 
-function ClearFilterButton({
-  labelsFilterVariable,
-  transitionsToFilterVariable,
-  transitionsFromFilterVariable,
-}: {
-  labelsFilterVariable: TextBoxVariable;
-  transitionsToFilterVariable: CustomVariable;
-  transitionsFromFilterVariable: CustomVariable;
-}) {
-  // get the current values of the filters
-  const valueInLabelsFilter = labelsFilterVariable.getValue();
-  //todo: use parsePromQLStyleMatcherLooseSafe to validate the label filter and check the lenghtof the result
-  const valueInTransitionsFilter = transitionsToFilterVariable.getValue();
-  const valueInTransitionsFromFilter = transitionsFromFilterVariable.getValue();
+export function ClearFilterButtonObjectRenderer({ model }: SceneComponentProps<ClearFilterButtonScenesObject>) {
+  // This make sure the component is re-rendered when the variables change
+  model.useState();
+
+  const labelsFilter = sceneGraph.interpolate(model, '${LABELS_FILTER}');
+  const stateTo = sceneGraph.interpolate(model, '${STATE_FILTER_TO}');
+  const stateFrom = sceneGraph.interpolate(model, '${STATE_FILTER_FROM}');
+
   // if no filter is active, return null
-  if (
-    !valueInLabelsFilter &&
-    valueInTransitionsFilter === StateFilterValues.all &&
-    valueInTransitionsFromFilter === StateFilterValues.all
-  ) {
+  if (!labelsFilter && stateTo === StateFilterValues.all && stateFrom === StateFilterValues.all) {
     return null;
   }
+
   const onClearFilter = () => {
-    labelsFilterVariable.setValue('');
-    transitionsToFilterVariable.changeValueTo(StateFilterValues.all);
-    transitionsFromFilterVariable.changeValueTo(StateFilterValues.all);
+    // eslint-disable-next-line
+    const labelsFiltersVariable = sceneGraph.lookupVariable(LABELS_FILTER, model) as TextBoxVariable;
```

Comment:
Should we try to use `instanceOf` operator rather than plain type casting to get more type safety?

Reply:
good idea!